tests/tt_metal/tt_metal/llk/test_single_core_matmul_int8.cpp

```diff
@@ -0,0 +1,230 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <chrono>
+#include <fmt/base.h>
+#include <gtest/gtest.h>
+#include <stdint.h>
+#include <tt-metalium/host_api.hpp>
+#include <tt-metalium/tt_metal.hpp>
+#include <unistd.h>
+#include <functional>
+#include <random>
+#include <map>
+#include <memory>
+#include <ostream>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#include <tt-metalium/bfloat16.hpp>
+#include <tt-metalium/buffer.hpp>
+#include <tt-metalium/buffer_types.hpp>
+#include <tt-metalium/circular_buffer_config.hpp>
+#include <tt-metalium/core_coord.hpp>
+#include <tt-metalium/data_types.hpp>
+#include "device_fixture.hpp"
+#include <tt-metalium/kernel_types.hpp>
+#include <tt-logger/tt-logger.hpp>
+#include <tt-metalium/program.hpp>
+#include <tt_stl/span.hpp>
+#include <tt-metalium/tt_backend_api_types.hpp>
+#include "tt_metal/test_utils/comparison.hpp"
+#include "tt_metal/test_utils/df/float32.hpp"
+#include "tt_metal/test_utils/packing.hpp"
+#include "tt_metal/test_utils/print_helpers.hpp"
+#include "tt_metal/test_utils/stimulus.hpp"
+
+namespace tt::tt_metal {
+
+using namespace tt;
+using namespace tt::test_utils;
+using namespace tt::test_utils::df;
+
+namespace unit_tests::compute::matmul {
+
+std::vector<int8_t> generate_uniform_int8(size_t num_elements, int8_t min, int8_t max, uint32_t seed = 0) {
+    std::vector<int8_t> random_array(num_elements);
+
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<int16_t> distrib(min, max);
+
+    for (int i = 0; i < num_elements; i++) {
+        random_array[i] = static_cast<int8_t>(distrib(gen));
+    }
```

Comment:
```suggestion
    for (auto& element : random_array) {
        element = static_cast<int8_t>(distrib(gen));
    }
```
Range-based for loop provides a neat way to populate the array with random numbers.

Or you could use `std::generate`:
```cpp
std::generate(random_array.begin(), random_array.end(), 
    [&]() { return static_cast<int8_t>(distrib(gen)); });
```